### PR TITLE
Auth(basic)

### PR DESCRIPTION
**Pull request by lannisterrr**
Added - Authentication
Added - basic form validation

### CODE REVIEW

File: src/components/RequiresAuth.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { useAuth } from '../contexts/auth-context';
+import { Navigate, useLocation } from 'react-router-dom';
+
+export default function RequiresAuth({ login, children }) {
```
Comment (junaidshaikh-js) [inline src/components/RequiresAuth.js]: Why are you passing `login` prop here?

File: src/contexts/auth-context.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      return {
+        ...state,
+        showPassword: false,
+        email: '',
+        password: '',
+        rememberMe: false,
+        signUpFirstName: '',
+        signUpLastName: '',
+        signUpEmail: '',
+        signUpPassword: '',
```
Comment (junaidshaikh-js) [inline src/contexts/auth-context.js:31]: You can destructure initial state here rather than passing all the values again manually

**Review by junaidshaikh-js — APPROVED** [review]
Good work. Clean code. 

Can be merged.

File: src/contexts/auth-context.js (hunk 2 above)
Comment (lannisterrr) [reply src/contexts/auth-context.js:31]: I have one state that I don't want to change that's why.

**Review by ashishnnnnn — APPROVED** [review]
Seems Good to merge.
Liked Your Validation Implementation..
Can be Merged :+1:

**Pull request merged**